src/queues/sendMessage.ts

```diff
@@ -17,15 +18,41 @@ declare global {
   }
 }
 
-const processJob: JobProcessor<typeof name> = () => async (job) => {
-  const { message, key } = job.data;
+const MESSAGE_LINES_LIMIT = 10;
+const MESSAGE_LENGTH_LIMIT = 2000;
+const breakDiscordMessage = (message: string) => {
+  if (message.length <= MESSAGE_LENGTH_LIMIT) return { current: message };
+  const lines = message.split('\n');
+  return {
+    current: lines.slice(0, MESSAGE_LINES_LIMIT).join('\n'),
+    rest: lines.slice(MESSAGE_LINES_LIMIT).join('\n'),
+  };
+};
+const sendDiscordMultipartMessage = async (
+  config: DiscordConfig,
+  message: string,
+  dispatchJobs: DispatchJobs,
+  key: ConfigKey,
+  replyToId?: string,
+) => {
+  const { current, rest } = breakDiscordMessage(message);
+  const msgId = await sendDiscordMessage(config, current, replyToId);
+  if (rest) {
+    await dispatchJobs([
+      { name: 'sendMessage' as const, data: { key, message: rest, replyToId: msgId } },
+    ]);
```

Comment:
might make more sense to return this and dispatch the job back in `processJob`.
```suggestion
    return { name: 'sendMessage' as const, data: { key, message: rest, replyToId: msgId } };
```